pkgs/development/misc/rappel/default.nix

```diff
@@ -0,0 +1,34 @@
+{ clang
+, fetchFromGitHub
+, libedit
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  name = "rappel";
+
+  src = fetchFromGitHub {
+    owner = "yrp604";
+    repo = "rappel";
+    rev = "95a776f850cf6a7c21923a2100b605408ef038de";
+    sha256 = "0fmd15xa6hswh3x48av4g1sf6rncbiinbj7gbw1ffvqsbcfnsgcr";
+  };
+
+  buildInputs = [ libedit ];
+  nativeBuildInputs = [ clang ];
+
+  makeFlags = "CC=clang";
+
+  installPhase = ''
+    make
+    cp -r bin $out/
+  '';
```

Comment:
Could you add `runHook preInstall` and `runHook postInstall` at the start and end of the `installPhase`?

Reply:
What the hooks suppose to do here?
Just curious.

Reply:
Added, but am curious about that too. Especially:

1) I expected the hooks to be implicitly run.
2) I do not do declare them. Do they have a default behavior?

Reply:
>     1. I expected the hooks to be implicitly run.

No, generally this is to enable for people to take the current derivation and add those phases manually. (which may be a little future proofing). Generally this is applicable when making something like `buildPythonPackage`, because many people want to run those hooks.
>     2. I do not do declare them. Do they have a default behavior?

They are empty by default, they are meant used by people using your package. I can only think of an override being useful here.